src/vs/platform/markers/common/problemMatcher.ts

```diff
@@ -781,6 +823,15 @@ class ProblemPatternParser extends Parser {
 	private validateProblemPattern(values: ProblemPattern[]): boolean {
 		let file: boolean, message: boolean, location: boolean, line: boolean;
 		let regexp: number = 0;
+		let patternsWithKind = values.filter((pattern, i) => {
```

Comment:
Could we fold this with the forEach from below. May be it would be easiest to turn the forEach into a for(pattern of values).